src/prefect/flows.py

```diff
@@ -158,6 +159,26 @@ def __init__(
             List[Callable[[schemas.core.Flow, FlowRun, State], None]]
         ] = None,
     ):
+        # Validate if hook passed is list and contains callables
+        hook_types = [on_completion, on_failure, on_cancellation, on_crashed]
+        hook_names = ["on_completion", "on_failure", "on_cancellation", "on_crashed"]
+        for hook, hook_name in zip(hook_types, hook_names):
+            if hook is not None:
+                try:
+                    hook = list(hook)
+                except TypeError:
+                    raise TypeError(
+                        f"Expected iterable for '{hook_name}'; got"
+                        f" {type(hook).__name__} instead."
+                    )
+
+                for hook_item in hook:
+                    if not callable(hook_item):
+                        raise TypeError(
+                            f"Expected callables in {hook}; got"
```

Comment:
I believe it would be more informative to include `hook_name` in this error message, to identify which hook category contains the non-callable item.

Reply:
Noted